vendor/src/github.com/docker/engine-api/types/container/config.go

```diff
@@ -5,6 +5,27 @@ import (
 	"github.com/docker/go-connections/nat"
 )
 
+// HealthConfig holds configuration settings for the HEALTHCHECK feature.
+type HealthConfig struct {
+	// Test to perform to check container is healthy.
+	// An empty slice means to inherit the default.
+	// Otherwise, Cmd[0] is "NONE" or "CMD".
+	Test strslice.StrSlice `json:",omitempty"`
+
+	// Nil means to inherit.
+	GracePeriod *float64 `json:",omitempty"` // Time after container start to assume failure means still starting.
```

Comment:
If you do `--health-grace=0` then you want a grace period of zero, which is not the same as not specifying the grace period (which inherits from the image).